app/controllers/assignments_controller.rb

```diff
@@ -45,106 +46,43 @@ def create
   end
 
   def edit
-    # give an error message is instructor have not set the time zone.
-    if current_user.timezonepref.nil?
-      flash.now[:error] = "You have not specified your preferred timezone yet. Please do this before you set up the deadlines."
-    end
-    @topics = SignUpTopic.where(assignment_id: params[:id])
-    @assignment_form = AssignmentForm.create_form_object(params[:id])
-    @user = current_user
+    current_user_timezonepref_nil_message
```

Comment:
It will be better the method name is a verb or an action, like `edit_params_setting`, instead of a noun.